data_steward/deid/parser.py

```diff
@@ -107,6 +107,17 @@ def query_priority(priority):
         raise ArgumentTypeError(message)
 
 
+def odataset_name_verification(odataset_name):
+    """
+    Ensures the output dataset name contains _deid
+    """
+    if "_deid" in odataset_name:
```

Comment:
This type check is a little too loose.  With this check, the string `any_random_deid_tag` would pass.  But it should fail because it doesn't end with `_deid`.